lib/cli/commands/walletwithdraw.ts

```diff
@@ -24,19 +24,17 @@ export const builder = (argv: Argv) => argv
     description: 'the fee in satoshis (or equivalent) per byte',
     type: 'number',
   })
-  .option('all', {
-    description: 'whether to withdraw all available funds',
-    type: 'boolean',
-  })
   .example('$0 walletwithdraw 0.1 BTC 1BitcoinEaterAddressDontSendf59kuE', 'withdraws 0.1 BTC')
   .example('$0 walletwithdraw 0.1 BTC 1BitcoinEaterAddressDontSendf59kuE 20', 'withdraws 0.1 BTC using 20 sats/byte')
-  .example('$0 walletwithdraw --all --currency BTC --address 1BitcoinEaterAddressDontSendf59kuE', 'withdraws all BTC');
+  .example('$0 walletwithdraw all BTC 1BitcoinEaterAddressDontSendf59kuE', 'withdraws all BTC');
 
 export const handler = async (argv: Arguments<any>) => {
+  argChecks.NUMBER_CHECK({ param: argv.amount, paramName: 'amount', allowedValues: ['all'] });
+
   const request = new WithdrawRequest();
   request.setCurrency(argv.currency.toUpperCase());
-  if (argv.all) {
-    request.setAll(argv.all);
+  if (argv.amount === 'all') {
+    request.setAll(true);
   } else {
     request.setAmount(coinsToSats(argv.amount));
```

Comment:
`coinsToSats` exepcts a `number` but we've defined `amount` as a `string` above so I'm concerned yargs would give us a string. I'm thinking we should do `parseFloat(argv.amount)` here.

Reply:
True that!